mlir/builder/ung.py

```diff
@@ -0,0 +1,81 @@
+"""A boiled down version of UniqueNameGenerator from inducer/pytools"""
```

Comment:
Please add a link to the GitHub repo as well.

Reply:
Done!